src/tools/generate/generate_chapters.js

```diff
@@ -51,6 +51,9 @@ const parse_file = async (markdown,chapter) => {
   const html = converter.makeHtml(markdown);
   let body = html;
 
+  // Make headers links
+  const header_regex = /<h([2-6]) id="(.*?)">(.*?)<\/h[2-6]>/gi;
+  body = body.replace(header_regex, "<h$1 id=\"$2\"><a href=\"#$2\" class=\"anchor-link\">$3</a></h$1>");
```

Comment:
Nit: "anchor-link" could be more descriptive, eg "heading-permalink"
Nit: mixed use of quotes in this file, use single quotes for consistency (and avoids the need to escape double quotes)

Reply:
I use the same class for figures. So heading-or-figure-permalink. Seems a bit long and not sure I agree that permalink is more descriptive (or accurate) than anchor-link tbh. So will leave as is unless you feel strongly about this.

Will fix the quotes.